pennylane/devices/default_clifford.py

```diff
@@ -519,58 +692,322 @@ def _measure_state(self, tableau_simulator, circuit, global_phase):
+            basis_states if len(qml.math.shape(basis_states)) > 1 else [basis_states]
+        )
+
+    # pylint: disable=too-many-branches, too-many-statements
+    def _measure_probability(self, meas, tableau_simulator, **kwargs):
+        """Measure the probability of each computational basis state."""
+        circuit, _ = kwargs.get("circuit"), tableau_simulator
+
+        # Obtain the measurement wires for getting the basis states
+        mobs_wires = meas.obs.wires if meas.obs else meas.wires
+        meas_wires = mobs_wires if mobs_wires else circuit.wires
+
+        if len(meas_wires) > 20 and self._prob_states is None and self._tableau:
+            raise ValueError(
+                "In order to maintain computational efficiency, \
+                with ``tableau=True``, the clifford device supports \
+                returning probability only for selected target \
+                computational basis states made of more than 20 qubits. \
+                Please use the `probability_target` property to set them."
+            )
+
+        if self._prob_states is None:
+            num_wires = len(meas_wires)
+            basis_vec = np.arange(2**num_wires)[:, np.newaxis]
+            self._prob_states = (((basis_vec & (1 << np.arange(num_wires)[::-1]))) > 0).astype(int)
+
+        # TODO: We might be able to skip the inverse done below
+        # (as the distribution should be independent of inverse)
+        diagonalizing_cit = kwargs.get("stim_circuit").copy()
+        diagonalizing_ops = [] if not meas.obs else meas.obs.diagonalizing_gates()
+        for diag_op in diagonalizing_ops:
+            # Check if it is Clifford
+            if diag_op.name not in _GATE_OPERATIONS:  # pragma: no cover
+                raise ValueError(
+                    f"Currently, we only support observables whose diagonalizing gates are Clifford, got {diag_op}"
+                )
+            # Add to the circuit to rotate the basis
+            stim_op = _pl_to_stim(diag_op)
+            if stim_op[0] is not None:
+                diagonalizing_cit.append_from_stim_program_text(f"{stim_op[0]} {stim_op[1]}")
+
+        # Build the Tableau simulator from the diagonalized circuit
+        circuit_simulator = stim.TableauSimulator()
+        circuit_simulator.do_circuit(diagonalizing_cit)
+        if not self._tableau:
+            state = self._measure_state(meas, circuit_simulator, circuit=circuit)
+            return meas.process_state(state, wire_order=circuit.wires)
+
+        tgt_states = self._prob_states
+        if not tgt_states.shape[1]:
+            raise ValueError("Cannot set an empty list of target states.")
+
+        if len(meas_wires) < tgt_states.shape[1]:
+            tgt_states = []
+            for state in self._prob_states:
+                if list(state[meas_wires]) not in tgt_states:
+                    tgt_states.append(list(state[meas_wires]))
+            tgt_states = np.array(tgt_states)
```

Comment:
what is the type of `state[meas_wires]` before casting to list? can we use `set()` construction instead to speed this up? no worries if it isn't trivial work